File: apps/gateway/src/loaders/container.loader.ts
import { createContainer } from "@zudojs/container";
import type { Container } from "@zudojs/container";
import type { Logger } from "@betng/service-kit";
import { LOGGER_TOKEN, UPSTREAM_CLIENTS_TOKEN } from "../constants/index.js";
import type { UpstreamClients } from "../interfaces/index.js";

export interface ContainerLoaderConfig {
  readonly clients: UpstreamClients;
  readonly logger: Logger;
}

export function loadContainer(config: ContainerLoaderConfig): Container {
  const container = createContainer();

  container.registerValue(UPSTREAM_CLIENTS_TOKEN, config.clients);
  container.registerValue(LOGGER_TOKEN, config.logger);

  return container.start();
}

File: services/risk/src/betng_risk/loaders/container_loader.py
from __future__ import annotations

import logging

from betng_service_kit import Container

from ..constants import AUDIT_RECORDER_TOKEN, LOGGER_TOKEN, RISK_REPOSITORY_TOKEN
from ..interfaces import AuditRecorder, RiskRepository


def load_container(
    repository: RiskRepository, audit: AuditRecorder, logger: logging.Logger
) -> Container:
    container = Container()

    container.register_value(RISK_REPOSITORY_TOKEN, repository)
    container.register_value(AUDIT_RECORDER_TOKEN, audit)
    container.register_value(LOGGER_TOKEN, logger)

    return container
